typescript/introduction/tsc-interfaces-classes/tsc-classes-intro.md

```diff
@@ -0,0 +1,77 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Classes Intro
+
+---
+## Content
+
+Classes represent the building blocks of the style of programming called *object-oriented (class-based) programming*, or OOP[1]. 
+
+The idea behind it is to organize related values and behaviors into definitions of entities that exist in your program (an "Animal", a "GamePlayer", a "User", etc). Then you can use these definitions, called classes, to *create (or instantiate)* those entities wherever you'd like. These instances of classes are called objects.
+
+To define a class, TypeScript gives us the `class` keyword. To create an instance of that class we can use the `new` keyword. Let's look at an example:
+
+```ts
+// define an Animal
+class Animal {
+  name: string;
+  constructor(animalName: string) {
+    this.greeting = animalName;
+  }
+  sayHi() {
+    console.log("Hi, I'm a " + this.greeting);
+  }
+}
+
+// create (instantiate) an Animal
+let zebra = new Animal("zebra");
+zebra.sayHi();
+// Hi, I'm a zebra
+```
+
+The benefit of classes is that they allow us to define data and behavior in one place, and then create many entities that use that data and follow that behavior as we see fit. In the example above, we used the `Animal` class to create a `zebra` but we could've also created a `tiger` and an `eagle` too. This allows us to have reusable code and not repeat ourselves.
+
+Note that using classes comes with some drawbacks as well[2]
+
+---
+## Practice
+
+Create an instance of the `City` class:
+
+```ts
+class City {
+  name: string;
+  constructor(cityName: string) {
+    this.name = cityName;
+  }
+  printCity() {
+    console.log(this.name);
+  }
+}
+
+let london = ??? ???('London')
+```
+
+* new
+* City
+* New
+* city
+
+---
+## Footnotes
+
+[1:Object-oriented programming] 
+OOP teaches us to organize data along with its associated behaviors into units called classes. Other classes can inherit this data/behaviors further and add their own.
```

Comment:
@sourabhv, related to [your comment](https://github.com/enkidevs/curriculum/pull/1819/files/78e43d2ba6a665799a5199ef305b2f52aa85103a#r415727153), how about we do this?

```suggestion
OOP teaches us to [organize](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)) data along with its associated behaviors into [units](https://en.wikipedia.org/wiki/Abstraction_(computer_science)) called classes. Other classes can [inherit](https://en.wikipedia.org/wiki/Inheritance_(object-oriented_programming)) this data/behaviors further and add their own and [related classes can be converted into one another](https://en.wikipedia.org/wiki/Polymorphism_(computer_science)).
```

Reply:
Looks good but `associated behaviors into [units] (https://en.wikipedia.org/wiki/Abstraction_(computer_science)`  but does this mean  abstraction? Also ~~this data/behaviors further and add their own~~ this data/behaviors further and add their own or change existing ones.